Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -4964,9 +4964,8 @@ void WebPage::willCommitLayerTree(RemoteLayerTreeTransaction& layerTransaction,
     layerTransaction.setPageExtendedBackgroundColor(page->pageExtendedBackgroundColor());
     layerTransaction.setSampledPageTopColor(page->sampledPageTopColor());
     if (std::exchange(m_needsFixedContainerEdgesUpdate, false)) {
-        auto fixedContainerEdges = frameView->fixedContainerEdges(sidesRequiringFixedContainerEdges());
-        protectedCorePage()->setLastTopFixedContainerColor(fixedContainerEdges.predominantColor(BoxSide::Top));
-        layerTransaction.setFixedContainerEdges(WTFMove(fixedContainerEdges));
+        page->updateFixedContainerEdges(sidesRequiringFixedContainerEdges());
+        layerTransaction.setFixedContainerEdges(FixedContainerEdges { page->fixedContainerEdges() });
```

Comment:
You can probably skip the type hint because the copy ctor isn't explicit

Reply:
Oh, I got an error with that approach:

```
In file included from […]/Build/Release/DerivedSources/WebKit/unified-sources/UnifiedSource137.cpp:2:
[…]/Source/WebKit/WebProcess/WebPage/WebPage.cpp:4968:49: error: binding reference of type 'WebCore::FixedContainerEdges' to value of type 'const FixedContainerEdges' drops 'const' qualifier
 4968 |         layerTransaction.setFixedContainerEdges({ page->fixedContainerEdges() });
In file included from […]/Build/Release/DerivedSources/WebKit/unified-sources/UnifiedSource137.cpp:2:
```

...but this made me realize that I could just make the parameter type a `const&` instead of `&&` (we end up needing a copy anyways, so this shouldn't be less efficient).

Thanks for the review!